scene/resources/motion_blur_compositor_effect.cpp

```diff
@@ -0,0 +1,262 @@
+
+	String overlay_glsl = R"(#[compute]
+#version 450
+
+layout(set = 0, binding = 0) uniform sampler2D blur_sampler;
+layout(rgba16f, set = 0, binding = 1) uniform image2D color_image;
+
+layout(local_size_x = 8, local_size_y = 8, local_size_z = 1) in;
+void main() {
+	ivec2 render_size = ivec2(textureSize(blur_sampler, 0));
+	ivec2 uv = ivec2(gl_GlobalInvocationID.xy);
+	if ((uv.x >= render_size.x) || (uv.y >= render_size.y)) {
+		return;
+	}
+	imageStore(color_image, uv, textureLod(blur_sampler, (vec2(uv) + 0.5) / render_size, 0.0));
+}
+)";
+	Ref<RDShaderFile> overlay_shader_file = memnew(RDShaderFile);
+	overlay_shader_file->parse_versions_from_text(overlay_glsl);
+	Ref<RDShaderSPIRV> overlay_bytecode = overlay_shader_file->get_spirv();
+	overlay_shader = RD::get_singleton()->shader_create_from_spirv(overlay_bytecode->get_stages(), "Motion Blur");
+	overlay_pipeline = RD::get_singleton()->compute_pipeline_create(overlay_shader);
+}
+
+RD::Uniform MotionBlurCompositorEffect::get_image_uniform(RID image, int binding) {
+	RD::Uniform uniform;
+	uniform.uniform_type = RD::UNIFORM_TYPE_IMAGE;
+	uniform.binding = binding;
+	uniform.append_id(image);
+	return uniform;
+}
+
+RD::Uniform MotionBlurCompositorEffect::get_sampler_uniform(RID image, int binding) {
+	RD::Uniform uniform;
+	uniform.uniform_type = RD::UNIFORM_TYPE_SAMPLER_WITH_TEXTURE;
+	uniform.binding = binding;
+	uniform.append_id(linear_sampler);
+	uniform.append_id(image);
+	return uniform;
+}
+
+void MotionBlurCompositorEffect::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("_render_callback", "effect_callback_type", "render_data"), &MotionBlurCompositorEffect::_render_callback);
+
+	ClassDB::bind_method(D_METHOD("get_motion_blur_samples"), &MotionBlurCompositorEffect::get_motion_blur_samples);
+	ClassDB::bind_method(D_METHOD("set_motion_blur_samples", "samples"), &MotionBlurCompositorEffect::set_motion_blur_samples);
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "motion_blur_samples"), "set_motion_blur_samples", "get_motion_blur_samples");
+
+	ClassDB::bind_method(D_METHOD("get_motion_blur_intensity"), &MotionBlurCompositorEffect::get_motion_blur_intensity);
+	ClassDB::bind_method(D_METHOD("set_motion_blur_intensity", "intensity"), &MotionBlurCompositorEffect::set_motion_blur_intensity);
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "motion_blur_intensity", PROPERTY_HINT_RANGE, "0,0.5,or_greater"), "set_motion_blur_intensity", "get_motion_blur_intensity");
+}
+
+void MotionBlurCompositorEffect::_render_callback(int p_effect_callback_type, const RenderDataRD *p_render_data) {
+	if (RD::get_singleton() && p_effect_callback_type != EFFECT_CALLBACK_TYPE_POST_TRANSPARENT) {
+		return;
+	}
+	Ref<RenderSceneBuffersRD> render_scene_buffers = p_render_data->get_render_scene_buffers();
+	RenderSceneData *render_scene_data = p_render_data->get_render_scene_data();
+	if (!(render_scene_buffers.is_valid() && render_scene_data)) {
```

Comment:
```suggestion
	if (render_scene_buffers.is_null() || !render_scene_data) {
```
Negated operations are usually harder to read than simpler forms